tracer/src/Datadog.Trace/Debugger/Sink/ProbeStatusSink.cs

```diff
@@ -78,9 +78,9 @@ public void AddProbeStatus(string probeId, Status status, Exception exception =
             _diagnostics.AddOrUpdate(probeId, timedMessage, (_, _) => (timedMessage));
             Enqueue(next);
 
-            bool ShouldOverwrite(Status currentStatus, Status nextStatus)
+            bool ShouldOverwrite(Status currentStatus, Status nextStatus, int currentProbeVersion, int nextProbeVersion)
             {
-                return nextStatus == Status.ERROR || currentStatus != nextStatus;
+                return nextStatus == Status.ERROR || currentStatus != nextStatus || currentProbeVersion < nextProbeVersion;
```

Comment:
nit: Perhaps rename `currentProbeVersion` to `oldProbeVersion` and `nextProbeVersion` to `newProbeVersion` for clarity?

Reply:
See https://github.com/DataDog/dd-trace-dotnet/pull/4853#discussion_r1390737034.